### PR TITLE
[Feature] Improve metadata model

### PR DESCRIPTION
**Pull request by hunter-cloud09**
Perfect metadata model

re #51

### CODE REVIEW

File: paimon-web-api/pom.xml [hunk 1]
```diff
@@ -63,6 +64,12 @@ under the License.
             <scope>test</scope>
         </dependency>
 
+        <dependency>
+            <groupId>com.alibaba</groupId>
+            <artifactId>fastjson</artifactId>
```
Comment (SteNicholas) [inline paimon-web-api/pom.xml:69]: Why use fastjson? IMO, fastjson has some security problem which is not recomeneded.

**Comment by hunter-cloud09** [comment]
@SteNicholas I have fixed them, PTAL.

File: paimon-web-server/src/main/java/org/apache/paimon/web/server/constant/MetadataConstant.java [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.paimon.web.server.constant;
+
+/** metadata constant. */
```
Comment (SteNicholas) [inline paimon-web-server/src/main/java/org/apache/paimon/web/server/constant/MetadataConstant.java:21]:
```suggestion
/** Metadata constant. */
```

File: paimon-web-server/src/main/java/org/apache/paimon/web/server/controller/MetadataController.java [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+@RequestMapping("/api/metadata")
+public class MetadataController {
+
+    private final MetadataService metadataService;
+
+    public MetadataController(MetadataService metadataService) {
+        this.metadataService = metadataService;
+    }
+
+    @PostMapping("/querySchemaInfo")
```
Comment (SteNicholas) [inline paimon-web-server/src/main/java/org/apache/paimon/web/server/controller/MetadataController.java:49]:
```suggestion
    @PostMapping("/query/schema")
```
BTW, why use `PostMapping`? It's better to use `GetMapping` for query request.

File: paimon-web-server/src/main/java/org/apache/paimon/web/server/controller/MetadataController.java [hunk 4]
```diff
@@ -0,0 +1,68 @@
...
+    public R<List<SchemaInfoVo>> getSchemaInfo(@RequestBody QueryMetadataInfoDto dto) {
+        return R.succeed(metadataService.getSchemaInfo(dto));
+    }
+
+    @PostMapping("/querySnapshotInfo")
```
Comment (SteNicholas) [inline paimon-web-server/src/main/java/org/apache/paimon/web/server/controller/MetadataController.java:54]: Ditto.

File: paimon-web-server/src/main/java/org/apache/paimon/web/server/controller/MetadataController.java [hunk 5]
```diff
@@ -0,0 +1,68 @@
...
+    public R<List<SnapshotInfoVo>> getSnapshotInfo(@RequestBody QueryMetadataInfoDto dto) {
+        return R.succeed(metadataService.getSnapshotInfo(dto));
+    }
+
+    @PostMapping("/queryManifestInfo")
```
Comment (SteNicholas) [inline paimon-web-server/src/main/java/org/apache/paimon/web/server/controller/MetadataController.java:59]: Ditto.

File: paimon-web-server/src/main/java/org/apache/paimon/web/server/controller/MetadataController.java [hunk 6]
```diff
@@ -0,0 +1,68 @@
...
+    public R<List<ManifestsInfoVo>> getManifestInfo(@RequestBody QueryMetadataInfoDto dto) {
+        return R.succeed(metadataService.getManifestInfo(dto));
+    }
+
+    @PostMapping("/queryDataFileInfo")
```
Comment (SteNicholas) [inline paimon-web-server/src/main/java/org/apache/paimon/web/server/controller/MetadataController.java:64]: Ditto.

File: paimon-web-server/src/main/java/org/apache/paimon/web/server/data/dto/QueryMetadataInfoDto.java [hunk 7]
```diff
@@ -0,0 +1,29 @@
...
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.paimon.web.server.data.dto;
+
+import lombok.Data;
+
+/** use for get table metadata. */
```
Comment (SteNicholas) [inline paimon-web-server/src/main/java/org/apache/paimon/web/server/data/dto/QueryMetadataInfoDto.java:23]:
```suggestion
/** DTO of query metadata is used for getting table metadata. */
```

File: paimon-web-server/src/main/java/org/apache/paimon/web/server/data/dto/QueryMetadataInfoDto.java [hunk 8]
```diff
@@ -0,0 +1,29 @@
...
+@Data
+public class QueryMetadataInfoDto {
```
Comment (SteNicholas) [inline paimon-web-server/src/main/java/org/apache/paimon/web/server/data/dto/QueryMetadataInfoDto.java:25]:
```suggestion
public class QueryMetadataDTO {
```

File: paimon-web-server/src/main/java/org/apache/paimon/web/server/data/model/MetadataFieldsModel.java [hunk 9]
```diff
@@ -0,0 +1,30 @@
...
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.paimon.web.server.data.model;
+
+import lombok.Data;
+
+/** table metadata fields model. */
```
Comment (SteNicholas) [inline paimon-web-server/src/main/java/org/apache/paimon/web/server/data/model/MetadataFieldsModel.java:23]:
```suggestion
/** Model of metadata fields. */
```

File: paimon-web-server/src/main/java/org/apache/paimon/web/server/data/vo/DataFileInfoVo.java [hunk 10]
```diff
@@ -0,0 +1,87 @@
...
+ * limitations under the License.
+ */
+
+package org.apache.paimon.web.server.data.vo;
+
+/** metadata data file return value. */
+public class DataFileInfoVo {
+
+    private final String partition;
+    private final Long bucket;
```
Comment (SteNicholas) [inline paimon-web-server/src/main/java/org/apache/paimon/web/server/data/vo/DataFileInfoVo.java:25]:
```suggestion
    private final long bucket;
```

File: paimon-web-server/src/main/java/org/apache/paimon/web/server/data/vo/SnapshotInfoVo.java [hunk 11]
```diff
@@ -0,0 +1,89 @@
...
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.paimon.web.server.data.vo;
+
+import java.time.LocalDateTime;
+
+/** metadata snapshot return value. */
```
Comment (SteNicholas) [inline paimon-web-server/src/main/java/org/apache/paimon/web/server/data/vo/SnapshotInfoVo.java:23]: Ditto.

File: paimon-web-server/src/main/java/org/apache/paimon/web/server/service/MetadataService.java [hunk 12]
```diff
@@ -0,0 +1,63 @@
...
+
+import org.apache.paimon.web.server.data.dto.QueryMetadataInfoDto;
+import org.apache.paimon.web.server.data.vo.DataFileInfoVo;
+import org.apache.paimon.web.server.data.vo.ManifestsInfoVo;
+import org.apache.paimon.web.server.data.vo.SchemaInfoVo;
+import org.apache.paimon.web.server.data.vo.SnapshotInfoVo;
+
+import java.util.List;
+
+/** metadata service. */
```
Comment (SteNicholas) [inline paimon-web-server/src/main/java/org/apache/paimon/web/server/service/MetadataService.java:29]:
```suggestion
/** Metadata service includes the service interfaces of metadata. */
```

File: paimon-web-server/src/main/java/org/apache/paimon/web/server/service/MetadataService.java [hunk 13]
```diff
@@ -0,0 +1,63 @@
...
+public interface MetadataService {
+
+    /**
+     * Retrieve a list of Metadata schema info.
+     *
+     * @param dto query metadata info
+     * @return a list of DatabaseInfo objects
+     */
+    List<SchemaInfoVo> getSchemaInfo(QueryMetadataInfoDto dto);
```
Comment (SteNicholas) [inline paimon-web-server/src/main/java/org/apache/paimon/web/server/service/MetadataService.java:38]: Remove the `info` from all interfaces of `MetadataService`.

File: paimon-web-server/src/main/java/org/apache/paimon/web/server/service/MetadataService.java [hunk 14]
```diff
@@ -0,0 +1,63 @@
...
+import org.apache.paimon.web.server.data.vo.SchemaInfoVo;
+import org.apache.paimon.web.server.data.vo.SnapshotInfoVo;
+
+import java.util.List;
+
+/** metadata service. */
+public interface MetadataService {
+
+    /**
+     * Retrieve a list of Metadata schema info.
```
Comment (SteNicholas) [inline paimon-web-server/src/main/java/org/apache/paimon/web/server/service/MetadataService.java:33]:
```suggestion
     * Retrieves a list of Metadata schema.
```

File: paimon-web-server/src/main/java/org/apache/paimon/web/server/service/impl/MetadataServiceImpl.java [hunk 15]
```diff
@@ -0,0 +1,217 @@
...
+import com.google.gson.reflect.TypeToken;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.stereotype.Service;
+
+import java.io.IOException;
+import java.util.LinkedList;
+import java.util.List;
+import java.util.Map;
+
+/** metadata service impl. */
```
Comment (SteNicholas) [inline paimon-web-server/src/main/java/org/apache/paimon/web/server/service/impl/MetadataServiceImpl.java:51]:
```suggestion
/** The implementation of {@link MetadataService}. */
```

File: paimon-web-server/src/test/java/org/apache/paimon/web/server/controller/MetadataControllerTest.java [hunk 16]
```diff
@@ -0,0 +1,209 @@
...
+import org.springframework.test.web.servlet.request.MockMvcRequestBuilders;
+import org.springframework.test.web.servlet.result.MockMvcResultHandlers;
+import org.springframework.test.web.servlet.result.MockMvcResultMatchers;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+/** test for MetadataController. */
```
Comment (SteNicholas) [inline paimon-web-server/src/test/java/org/apache/paimon/web/server/controller/MetadataControllerTest.java:46]:
```suggestion
/** Tests for {@link MetadataController}. */
```

**Review by SteNicholas — CHANGES_REQUESTED** [review]
@hunter-cloud09, thanks for updates. I left some comments for the updates. PTAL.

**Comment by hunter-cloud09** [comment]
@SteNicholas Fixed, PTAL.

File: paimon-web-server/src/main/java/org/apache/paimon/web/server/data/dto/QueryMetadataDTO.java [hunk 17]
```diff
@@ -0,0 +1,29 @@
...
+ */
+
+package org.apache.paimon.web.server.data.dto;
+
+import lombok.Data;
+
+/** DTO of query metadata is used for getting table metadata. */
+@Data
+public class QueryMetadataDTO {
+    private String catalogName;
```
Comment (SteNicholas) [inline paimon-web-server/src/main/java/org/apache/paimon/web/server/data/dto/QueryMetadataDTO.java:26]: Adds the new line among fields.

File: paimon-web-server/src/main/java/org/apache/paimon/web/server/data/model/MetadataFieldsModel.java [hunk 18]
```diff
@@ -0,0 +1,30 @@
...
+ */
+
+package org.apache.paimon.web.server.data.model;
+
+import lombok.Data;
+
+/** Model of metadata fields. */
+@Data
+public class MetadataFieldsModel {
+    private int id;
```
Comment (SteNicholas) [inline paimon-web-server/src/main/java/org/apache/paimon/web/server/data/model/MetadataFieldsModel.java:26]: Ditto.

File: paimon-web-server/src/main/java/org/apache/paimon/web/server/data/vo/DataFileVO.java [hunk 19]
```diff
@@ -0,0 +1,87 @@
...
+
+    public String getFilePath() {
+        return filePath;
+    }
+
+    public String getFileFormat() {
+        return fileFormat;
+    }
+
+    public static Builder builder() {
```
Comment (SteNicholas) [inline paimon-web-server/src/main/java/org/apache/paimon/web/server/data/vo/DataFileVO.java:52]: Could all VOs add the setter instead of builder?

**Review by SteNicholas — COMMENTED** [review]
@hunter-cloud09, there are some style that could be improved. PTAL.

**Review by SteNicholas — APPROVED** [review]
LGTM.

**Review by SteNicholas — APPROVED** [review]
LGTM.

**Pull request merged**